src/service/resource.js

```diff
@@ -396,7 +412,12 @@ export class Resource {
    * @return {!../layout-rect.LayoutRectDef}
    */
   getLayoutBox() {
-    return this.layoutBox_;
+    if (!this.isFixed_) {
+      return this.layoutBox_;
+    }
+    const viewport = this.resources_.getViewport();
+    return moveLayoutRect(this.layoutBox_, viewport.getScrollLeft(),
```

Comment:
> The definition of the getLayoutBox is that it doesn't change based on the scroll state.

I've taken it to mean "give me the absolute position from the origin (0, 0)", which is how everything (until we [introduced `#isFixed`](https://github.com/ampproject/amphtml/pull/3429)) treated it. This just fixes the value fixed-position's return.

> 1. Modify overlaps method to incorporate the knowledge of isFixed. 

The overlaps stuff is totally superficial to the PR. I need to fix `#getLayoutBox` for Flying Carpet, which depends on it returning an absolute position.

>  I don't know who uses it and where. I think it's best to replace this call with getLayoutSize() or getLayoutWidth/Height pair.

I'd love to see this. Should we wait for the Layers work?